default.nix

```diff
@@ -31,7 +31,7 @@ let
   inherit (llvm-backend-project) clang llvm-backend;
 
   k = callPackage ./nix/k.nix {
```

Comment:
Can you give me some details about what `callPackage` does?

Reply:
`callPackage` calls a function in another file with default arguments, like this:

1. `callPackage` `import`s the file (first argument). The file should be a Nix function taking a set of arguments (`{a, b, ...}:`).
2. `callPackage` applies the argument set (second argument) to the imported function.
3. If any of the arguments of the imported function are still missing, `callPackage` looks them up in `pkgs`.

Reply:
Now I see it's presented in the 13th Nix Pill, but after the 12th I thought I don't have a good enough understanding even though I completed the exercises, so I started reading Eelco Dolstra's thesis. Maybe I should resume reading the pills, now that I read some chapters from the thesis.

Reply:
I don't know... I've never read his thesis myself. I know that some of the internal details are only documented ther, but I don't know if it's necessary to understand the internals.